Add option to link against non-catkin librealsense

rojkov opened the pull request:
In OpenEmbedded setups where both `meta-ros` and `meta-intel-realsense` layers are used it's problematic to avoid two copies of librealsense installed onto an embedded target: one catkit-based installed under `/opt/ros` and one other installed to a standard system location.

This patch adds an option making realsense_camera link against librealsense provided by the host system.

By default the currently existing behavior is preserved.

Commit:
Add option to link against non-catkin librealsense

In OpenEmbedded setups where both meta-ros and
meta-intel-realsense layers are used it's problematic
to avoid two copies of librealsense installed onto
an embedded target: one catkit-based installed under
/opt/ros and one other installed to a standard system
location.

This patch adds an option making realsense_camera link
against librealsense provided by the host system.

By default the currently existing behavior is preserved.

Signed-off-by: Dmitry Rozhkov <dmitry.rozhkov@linux.intel.com>

rojkov commented:
ping.
The corresponding meta-ros PR  is here https://github.com/bmwcarit/meta-ros/pull/457. And it has been tested already.

mdhorn reviewed (changes requested):
Sorry, we were in the middle of a regression which needed to be completed before merging new code.

I also want to try a few test builds to approve possible fails in the ROS Build farm.

The code looks okay to me, but please rebase this branch as there have been minor updates.

Commit:
Enable roslint when CATKIN_ENABLE_TESTING is True

roslint depends on catkin_run_tests_target() which is defined only
when CATKIN_ENABLE_TESTING is True.

Thus run roslint functions in CMakeLists.txt only when testing
is enabled.

Signed-off-by: Dmitry Rozhkov <dmitry.rozhkov@linux.intel.com>

rojkov commented:
* rebased to the current indigo-devel
* made running roslint tasks conditional since they require `CATKIN_ENABLE_TESTING=ON`
* force pushed.

fmessmer commented:
Somehow https://github.com/intel-ros/realsense/commit/701f9393bf102c5d130fbe80b434b4060c405c31 lets our Travis CI fail due to 
```
Errors     << realsense_camera:cmake /root/catkin_ws/logs/realsense_camera/build.cmake.000.log

CMake Error at /root/catkin_ws/src/realsense/realsense_camera/CMakeLists.txt:122 (roslint_cpp):

  Unknown CMake command "roslint_cpp".
```
see https://travis-ci.org/ipa320/cob_robots/jobs/212986198#L6578

our travis.yml can be seen here: https://github.com/ipa320/cob_robots/blob/indigo_dev/.travis.yml

---
Submitting an issue seems to be blocked somehow...

mdhorn commented:
There appears to be some race condition in the change.
`catkin_make_isolated` also has the same fail signature, where `catkin_make_isolated -j1` passes.
I'll be submitting a PR with a minor change to fix this.
